WBooks/AddNew/AddNewViewController.swift

```diff
@@ -23,11 +23,110 @@ class AddNewViewController: UIViewController {
         super.viewDidLoad()
+        setupActions()
         navigationController?.navigationBar.topItem?.title = NSLocalizedString("TITLE_ADDNEW", comment: "")
     }
     
     override func loadView() {
         view = addNewView
     }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        addNewView.submitButton.withGradient()
+        setupTextField()
+    }
+    
+    func setupTextField(){
+        addNewView.bookNameTextField.config()
+        addNewView.authorTextField.config()
+        addNewView.yearTextField.config()
+        addNewView.topicTextField.config()
+        addNewView.descriptionTextField.config()
+    }
+    
+    func setupActions(){
+        addNewView.addImageButton.addTarget(self, action: #selector(selectImage), for: .touchUpInside)
+        addNewView.submitButton.addTarget(self, action: #selector(validateAddBook), for: .touchUpInside)
+    }
+    
+    @objc func selectImage(){
+        let alertController = UIAlertController(title: .none, message: .none, preferredStyle: .actionSheet)
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.delegate = self
+        let chooseAction = UIAlertAction(title: NSLocalizedString("OPTION_GALLERY", comment: ""), style: .default) { _ in
+            imagePickerController.sourceType = .photoLibrary
+            self.present(imagePickerController, animated: true, completion: .none)
+        }
+        alertController.addAction(chooseAction)
+        if UIImagePickerController.isSourceTypeAvailable(.camera) {
+            let takeAction = UIAlertAction(title: NSLocalizedString("OPTION_CAMERA", comment: ""), style: .default) { _ in
+                imagePickerController.sourceType = .camera
+                self.present(imagePickerController, animated: true, completion: .none)
+            }
+            alertController.addAction(takeAction)
+        }
+        let cancelAction = UIAlertAction(title: NSLocalizedString("OPTION_CANCEL", comment: ""), style: .cancel, handler: .none)
+        alertController.addAction(cancelAction)
+        self.present(alertController, animated: true, completion: nil)
+    }
+    
+    @objc func validateAddBook() {
+        if isEmpyForm() {
+            addNewViewModel.setParametersBook(
+                title: addNewView.bookNameTextField.text ?? "",
+                author: addNewView.authorTextField.text ?? "",
+                genre: addNewView.topicTextField.text ?? "",
+                year: addNewView.yearTextField.text ?? "")
+            addBook()
+        } else {
+            self.showAlert(title: NSLocalizedString("TITLE_ERROR", comment: ""),
+                           message: NSLocalizedString("ERROR_EMPY_TEXTFLIED", comment: ""),
+                           buttonDone: NSLocalizedString("BUTTON_DONE", comment: ""))
+        }
+    }
+    
+    func addBook()  {
+        addNewViewModel.addNewBook() { error in
+            if error == nil {
+                self.cleanForm()
```

Comment:
Move cleanForm above if/esle statement to don't reapeat yourself (DRY principle)

Reply:
Done